### ISSUE TITLE
Controlling the visibility of the struct generated by the #[dbus_proxy] macro

### ISSUE THREAD

**Issue by zeenix** [dbus2/zbus#129]
In GitLab by @mominul on Jan 23, 2021, 18:01

Currently the struct and its members generated by the `#[dbus_proxy]` macro are `pub` by default.
I think it'd be great if we could control the visibility at least of the generated struct.
The macro would look for a visibility modifier in the trait declaration.
```
// It will be a public struct.
#[dbus_proxy(interface = "...")]
pub trait Interface {
    //...
}

// It will be a internal struct.
#[dbus_proxy(interface = "...")]
trait Interface {
    //...
}

// It will be a public in crate struct.
#[dbus_proxy(interface = "...")]
pub(crate) trait Interface {
    //...
}
```

Thanks!

**Comment by SludgePhD** [comment]
This would be great to have. I'm currently trying to use zbus as a purely internal dependency and all the `pub` types it generates are making this more difficult than I'd hoped.

**Comment by zeenix** [comment]
This was fixed by cd7c61f339dc9face6023733d17efd4f0b301728 and da69867603433ae58f59d5b6651b2f8e60f484fc.
